llvm/lib/Target/AMDGPU/AMDGPUTargetMachine.cpp

```diff
@@ -821,8 +825,15 @@ void AMDGPUTargetMachine::registerPassBuilderCallbacks(PassBuilder &PB) {
           PM.addPass(AMDGPUSwLowerLDSPass(*this));
         if (EnableLowerModuleLDS)
           PM.addPass(AMDGPULowerModuleLDSPass(*this));
-        if (EnableAMDGPUAttributor && Level != OptimizationLevel::O0)
-          PM.addPass(AMDGPUAttributorPass(*this));
+        if (Level != OptimizationLevel::O0) {
+          if (EnableAMDGPUAttributor)
+            PM.addPass(AMDGPUAttributorPass(*this));
+          // Do we really need internalization in LTO?
+          if (InternalizeSymbols) {
```

Comment:
This needs to be moved before attributor.

Reply:
Do we need the custom internalize anymore? I thought this was because of mis-set visibility in the libraries, but that was fixed?

Reply:
I'm not sure. I can start a PSDB session to test it out whether it makes difference after removing it.

Reply:
Well, we probably still need that for those don't use LTO, such as `comgr`.